### PR TITLE
chore: support node deps in local env manager

### PR DESCRIPTION
**Pull request by XiaofuHuang**
1. Support node deps checker
2. Remove call depsManager

Detail: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/12863537

### CODE REVIEW

File: packages/fx-core/src/common/local/localEnvManager.ts [hunk 1]
```diff
@@ -37,38 +34,38 @@ export class LocalEnvManager {
...
-      fastFail: true,
-    });
+    // Function core tool
+    if (includeBackend) {
+      depsTypes.push(DepsType.FuncCoreTools);
+    }
+
+    // Ngrok
+    if (includeBot) {
+      depsTypes.push(DepsType.Ngrok);
```
Comment (kuojianlu) [inline packages/fx-core/src/common/local/localEnvManager.ts:65]: What if skipping ngrok?

File: packages/fx-core/src/common/local/localEnvManager.ts [hunk 2]
```diff
@@ -37,38 +34,38 @@ export class LocalEnvManager {
...
+    if (includeFrontend) {
+      depsTypes.push(DepsType.Dotnet);
+    }
 
-    return await depsManager.ensureDependencies(this.getValidDeps(projectSettings), {
-      fastFail: true,
-    });
+    // Function core tool
+    if (includeBackend) {
+      depsTypes.push(DepsType.FuncCoreTools);
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply packages/fx-core/src/common/local/localEnvManager.ts:60]: backend extension install now is not in DepsType

File: packages/fx-core/src/common/local/localEnvManager.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/fx-core/src/common/local/localEnvManager.ts:65]: In the `Inner Loop New Workflow`,  
> localSettings.json 
> Remove frontend/trustDevCert (moved to vsc settings) 
> Remove bot/skipNgrok (moved to vsc settings) 

Can we remove it from here?

File: packages/fx-core/src/common/local/localEnvManager.ts [hunk 3]
```diff
@@ -37,38 +34,38 @@ export class LocalEnvManager {
...
 
-  public async checkAndResolveDependencies(
-    projectSettings: ProjectSettings
-  ): Promise<DependencyStatus[]> {
-    const depsLogger = new DepsLoggerAdapter(this.logger);
-    const depsTelemetry = new DepsTelemetryAdapter(this.telemetry);
-    const depsManager = new DepsManager(depsLogger, depsTelemetry);
+    // Dotnet
+    if (includeFrontend) {
+      depsTypes.push(DepsType.Dotnet);
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply packages/fx-core/src/common/local/localEnvManager.ts:55]: Thanks! It has been changed.

File: packages/fx-core/src/common/local/localEnvManager.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kuojianlu) [reply packages/fx-core/src/common/local/localEnvManager.ts:61]: Understood, thanks.

**Review by kuojianlu — APPROVED** [review]

File: packages/fx-core/src/common/local/localEnvManager.ts (hunk 1 above)
Comment (swatDong) [reply packages/fx-core/src/common/local/localEnvManager.ts:66]: The setting may be moved to another place but still configurable. So need to consider `skipNgrok` flag.

Comment (XiaofuHuang) [reply packages/fx-core/src/common/local/localEnvManager.ts:66]: Now all the vsc settings are loaded in vscodeChecker.
```
private async isEnabled(dep: DepsType): Promise<boolean> {
    switch (dep) {
      case DepsType.AzureNode:
      case DepsType.SpfxNode:
        return checkerEnabled(VSCodeDepsChecker.validateNodeVersionKey);
      case DepsType.FunctionNode:
        return checkerEnabled(VSCodeDepsChecker.validateNodeVersionKey) && (await hasFunction());
      case DepsType.Dotnet:
        return checkerEnabled(VSCodeDepsChecker.validateDotnetSdkKey);
      case DepsType.FuncCoreTools:
        return checkerEnabled(VSCodeDepsChecker.validateFuncCoreToolsKey) && (await hasFunction());
      case DepsType.Ngrok:
        // TODO: remove hasBot and get hasNgrok from workspace configuration
        return (await hasBot()) && (await hasNgrok());
      default:
        return false;
    }
```
Can we only generate the deps types according to projectSettings in localEnvManager?

Comment (swatDong) [reply packages/fx-core/src/common/local/localEnvManager.ts:66]: I see, so this is to generate the deps list and still need `vscodeChecker`/`cliChecker` to double check all flags.

**Review by swatDong — APPROVED** [review]

**Pull request merged**